core/unit_test/sycl/TestSYCL_InterOp_Graph.cpp

```diff
@@ -14,16 +14,20 @@
 //
 //@HEADER
 
-#include <filesystem>
-#include <fstream>
-#include <regex>
-
 #include <TestSYCL_Category.hpp>
 #include <Kokkos_Core.hpp>
 #include <Kokkos_Graph.hpp>
 
 #include <gtest/gtest.h>
 
+#include <fstream>
+#include <regex>
+
+#if !(defined(_GLIBCXX_RELEASE) && _GLIBCXX_RELEASE < 9) && \
+    !(defined(_LIBCPP_VERSION) && _LIBCPP_VERSION < 110000)
+#include <filesystem>
+#endif
```

Comment:
Trying to include `filesystem` caused problems with the underlying `clang` version used for `clang-tidy`.

Reply:
Ok

Reply:
I am surprised we are using an old clang here, doesn't it come from the oneapi install ? I would expect a recent clang/clang-tidy ?

Reply:
I could still use an older stdlibc++. I haven't investigated further, though.

Reply:
I didn't see that the file had already such `ifdef` logic in the tests